ADR/0033-upstreaming-integration-service.md

```diff
@@ -0,0 +1,270 @@
+    name: '.metadata.labels."appstudio.openshift.io/component"'
+    source:
+      git:
+        revision: '.metadata.labels."pipelinesascode.tekton.dev/sha"'
+        url: '.metadata.labels."pipelinesascode.tekton.dev/repo-url"'
+```
+
+Don't take the fields literally. The details will likely change after the ADR has settled. It is
+here for illustration purposes.
+
+The `selector` field will be used to identify resources that should trigger the construction of new
+[TestSubjects]. When created, the *basis* for the new test subject should always be taken from the
+**control** [TestSubject]. The values extracted from the triggering resource by way of the
+expressions on the `extractor` resource should be applied to that basis in the construction of a new
+[TestSubject]. If there is no **control** [TestSubject], then the new TestSubject should be created
+containing only one element: the new pullspec.
+
+In the above example, the constructor will select only pipelineruns which are completed and which
+are associated with the app `my-application`. When triggered, it will extract the same image and
+source details that we extract today to supply to a new [TestSubject].
+
+### Release Initiation, as owned by the release-service
+
+For the purpose of this ADR, we will only say that release-service will take over the
+responsibility of initiating Releases. That will be described in more detail in a subsequent,
+forthcoming ADR.
+
+Note, in this pattern, [Release] construction looks a lot like [TestSubject] construction. The
+triggered system knows enough about the triggering system to optionally initiate itself.
+
+## Use Cases in Detail
+
+**Using integration-service as a remote API from Jenkins.** A hypothetical platform team might want
+to make use of [integration-service] from a jenkins pipeline. For example, perhaps their jenkins job
+performs CI on a git repository that contains pullspecs which should be mirrored to some environment
+of theirs. Before mirroring, they want to make sure that the same test works against dozens of
+different kubernetes configurations on different public clouds. That platform team first
+creates a single tekton pipeline that contains the logic to perform the test given a single cluster.
+They then configure dozens of [IntegrationTestScenario] resources, one for each kubernetes
+configuration they want to target. When a new set of pullspecs is proposed to their git repo,
+Jenkins responds. Jenkins then creates a single [TestSubject] in the kubernetes namespace,
+representing the new set of images under test. [integration-service] responds, inspecting all of the
+defined [IntegrationTestScenarios], it invokes the [deployment-target-operator] (from [Decoupling
+Deployment]) to provision each cluster configuration and it runs tests of the [TestSubject] against
+each. The aggregate results are stored in the `status` of the [TestSubject] for jenkins to monitor
```

Comment:
This effectively provides a way for some "arbitrary" process to provision an environment and run a workload on it. This feels like a dangerous attack vector.

Reply:
Ah - the process must have RBAC to create TestSubjects. That's the limiting factor.

Reply:
.. similar to RBAC which limits which accounts can create DeploymentTargetClaims, which accomplishes a similar thing more directly.